app/cache/cache-service.js

```diff
@@ -0,0 +1,44 @@
+const NodeCache = require('node-cache');
+const { Logger } = require('@hmcts/nodejs-logging');
+const logger = Logger.getLogger('CacheService');
+
+class Cache {
+
+  constructor(name, ttlSeconds, checkPeriodSeconds) {
+    this.cache = new NodeCache({ stdTTL: ttlSeconds, checkperiod: checkPeriodSeconds });
+    this.name = name;
+  }
+
+  get(key) {
+    return this.cache.get(key);
+  }
+
+  set(key, value) {
+    this.cache.set(key, value);
+  }
+
+  /**
+   * If given key is already in the cache, returns associated value.
+   * Otherwise, computes value from given expression op, stores with key in the cache and returns that value.
+   */
+  getOrElseUpdate(key, op) {
+    const value = this.get(key);
+    if (value) {
+      return Promise.resolve(value);
+    }
+
+    return op().then(result => {
+      this.set(key, result);
+      return result;
+    }).catch(error => {
+      logger.warn(`Error in store function for cache '${this.name}'`);
```

Comment:
```suggestion
      logger.warn(`Error computing value to be cached for cache '${this.name}'`);
```

Reply:
shall we also log the key?

Reply:
FYI: I have just stopped RDM-8304 (hmcts/ccd-case-activity-api#192) from logging the key as this may contain part of the token.  I think @danlysiak has already done the same to this PR.